core-rust/mesh-api-server/src/mesh_api/handlers/mempool_transaction.rs

```diff
@@ -55,11 +53,11 @@ pub(crate) async fn handle_mempool_transaction(
 
         UserTransaction::V2(_) => {
             return Err(ResponseError::from(ApiError::InvalidTransaction)
-                .with_details(format!("Transactions V2 not supported")))
+                .with_details(format!("Transaction V2 not supported")))
         }
     };
 
-    let database = state.state_manager.database.snapshot();
+    let database = state.state_manager.database.access_direct();
```

Comment:
Again, let's explain why if this is safe.

Reply:
Decided to revert to `snapshot()` as per [this conversation](https://rdxworks.slack.com/archives/C071C6DB8TG/p1733841098517859?thread_ts=1732877188.486819&cid=C071C6DB8TG)